### PR TITLE
Improve rabbitmq transport logging on connection errors

### PR DESCRIPTION
**Pull request by moises-silva**
This should fix issue #30

### CODE REVIEW

**Comment by cwjohnston** [comment]
Closes #30

**Comment by portertech** [comment]
@moises-silva @cwjohnston I would love to see log lines like https://github.com/sensu/sensu/blob/master/lib/sensu/daemon.rb#L256, using "transport connection error" as the message, then use `:error` or `:reason` as context keys.

**Comment by portertech** [comment]
Otherwise these log events lack context.

**Comment by cwjohnston** [comment]
@portertech so far this lib hasn't pulled in sensu-logger for testing, using Ruby's built-in Logger class instead. It seems like `Logger` doesn't accept the same `data` hash argument Sensu::Logger uses for providing additional context. Should we change the tests to bring in Sensu::Logger?

**Comment by portertech** [comment]
@cwjohnston yes, let's pull it in without a version constraint.

File: lib/sensu/transport/rabbitmq.rb [hunk 1]
```diff
@@ -185,18 +186,34 @@ def next_connection_options
...
+        @connection.on_tcp_connection_loss do
+          @logger.warn("transport connection error", :reason => "TCP connection lost")
+          reconnect
+        end
+        @connection.on_skipped_heartbeats do
+          @logger.warn("transport connection error", :reason => "Skipped heartbeats")
+          reconnect
+        end
+        @connection.on_closed do
+          @logger.debug("connection closed")
```
Comment (portertech) [inline lib/sensu/transport/rabbitmq.rb]: "transport connection closed"

File: lib/sensu/transport/rabbitmq.rb [hunk 2]
```diff
@@ -185,18 +186,34 @@ def next_connection_options
...
+        @connection.on_tcp_connection_loss do
+          @logger.warn("transport connection error", :reason => "tcp connection lost")
+          reconnect
+        end
+        @connection.on_skipped_heartbeats do
+          @logger.warn("transport connection error", :reason => "skipped heartbeats")
+          reconnect
+        end
+        @connection.on_closed do
+          @logger.debug("connection closed")
```
Comment (portertech) [inline lib/sensu/transport/rabbitmq.rb]: "transport connection closed"

File: lib/sensu/transport/rabbitmq.rb [hunk 3]
```diff
@@ -185,18 +186,34 @@ def next_connection_options
 
       def setup_connection(options={})
         reconnect_callback = Proc.new { reconnect }
+        on_possible_auth_failure = Proc.new {
+          @logger.warn("transport connection error", :reason => "possible authentication failure. wrong credentials?")
```
Comment (portertech) [inline lib/sensu/transport/rabbitmq.rb]:
```
@logger.warn("transport connection error", {
  :reason => "possible authentication failure. wrong credentials?"),
  :user =>  options[:user]
}
```
And remove the debug log event below?

Comment (portertech) [inline lib/sensu/transport/rabbitmq.rb]: We already debug log authentication information in core https://github.com/sensu/sensu/blob/master/lib/sensu/daemon.rb#L248-L251

**Comment by portertech** [comment]
Fantastic!

**Comment by cwjohnston** [comment]
@portertech disregarding the one failure here, i think this is good to merge.

I believe the [single failed platform in question](https://travis-ci.org/sensu/sensu-transport/jobs/170646947) relates to my accidentally pushing (and subsequently deleting) the branch of same name on this repo. I had meant to push to sangoma/sensu-transport fork to update this pull request, and I guess travis decided they would be the same PR? 🤔

**Comment by portertech** [comment]
@cwjohnston ok, going to merge this, we can address the build failures if they continue on jruby.

**Pull request merged**